eem-sparks/src/main/java/com/nice/eem/sparkathon/AWSS3Service.java

```diff
@@ -0,0 +1,68 @@
+package com.nice.eem.sparkathon;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+
+public class AWSS3Service {
+
+    /**
+     * uploading a file to s3 bucket
+     */
+    public static void uploadFile(String bucketName, String keyName, String filePath) {
+        try {
+            AmazonS3 s3Client = AmazonS3ClientBuilder.defaultClient();
+            s3Client.putObject(bucketName, keyName, filePath);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+    /**
+     * downlnoading a file from s3 bucket
+     */
+    public static void downloadFile(String bucketName, String keyName, String filePath) {
+         try {
+            AmazonS3 s3Client = AmazonS3ClientBuilder.defaultClient();
+            s3Client.getObject(bucketName, keyName);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * listing all the files in a bucket
+     */
+
+    public static void listFiles(String bucketName) {
+        try {
+            AmazonS3 s3Client = AmazonS3ClientBuilder.defaultClient();
+            s3Client.listObjects(bucketName);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code is manually checking for the existence of a bucket in S3. You could use the SDK (operation name: [doesBucketExist](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/AmazonS3Client.html#doesBucketExist-java.lang.String-) or [doesBucketExistV2](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/AmazonS3Client.html#doesBucketExistV2-java.lang.String-)) to automatically check for this instead.